src/main/java/eu/deltasource/audioplayer/Application.java

```diff
@@ -0,0 +1,138 @@
+package eu.deltasource.audioplayer;
+
+//Тема 2: Audio Player XL2
+//        Ще направим същата задача като `Задача 1`, но ще създадем допълнителен клас `Song`.
+//        В конструкторите, задайте стойности по подразбиране (вие изберете какви). Избирайте
+//        подходящи имена на методите, както и подходящи типове на резултатите. Създайте клас
+//        `Application`, който трябва да е стартовия файл на програмата ни.
+//        Не забравяйте да оставяте коментари!
+//
+//        1. Създайте клас `Song`. Създайте няколко член-променливи, като прецените сами какъв
+//        тип трябва да бъдат
+//        - `title` - заглавие на песента
+//        - `author` - автор
+//        - `genre` - жанр
+//        - `timing` - времетраене на песента
+//        Напишете метод, който да извежда в конзолата характеристиките на класа.
+//        Напишете метод, който да извежда в конзолата форматирано заглавието на песента и
+//        автора.
+//
+//        2. Създайте клас `AudioPlayer`. Създайте няколко instance променливи, като прецените
+//        сами какъв тип трябва да бъдат
+//        - `songs` - всички песни
+//----------------------------------------------------------------------------------
+//        Напишете метод, който да извежда в конзолата характеристиките на класа.
+
+//        Напишете метод, който да стартира изпълнението на песните последователно.
+
+//        Напишете метод, който да спира изпълнението на песните и след последващо
+//        стартиране да започват от начало.
+
+//        Напишете метод, който да прави пауза на текущата песен и след последващото
+//        стартиране да продължава от същата песен.
+
+//        Напишете метод, който трябва да стартира следваща песен.
+
+//        Напишете метод, който трябва да стартира предишна песен.
+
+//        Напишете метод, който трябва да изпълнява песните разбъркано.
+
+//        Напишете метод, който трябва да изписва в конзолата коя подред е песента, която в
+//        момента се изпълнява, както и нейните заглавие и изпълнител.!!!!!!!!!!!!!
+//--------------------------------------------------------------------------------------
+
+//--------------------------------------------------------------------------------
+//        Напишете метод, който трябва да търси по заглавие и да извежда в конзолата името на
+//        изпълнителя, ако е намерено съвпадение и коя подред е.
+
+//        Напишете метод, който трябва да търси по изпълнител и да извежда в конзолата всички
+//        негови песни.
+//        Напишете метод, който трябва да извежда в конзолата броят на всички песни.
+//        Напишете метод, който може да добавя нова песен.
+//        Напишете метод, който може да премахва песен.
+//-----------------------------------------------------------------------------------------
+
```

Comment:
Javadoc everywhere, you have some more complext logic and constructions at places which need explanation.
You're making your code more complex by using some "patterns", without actually getting the benefit of using them. For example if you don't need to reuse the common code of something, don't force inheritance. (Song -> AudioPlayable || Player -> AudioPlayer abstract)